### ISSUE TITLE
Support max_parallel_api_rate to call InvokeModel API parallelly to shorten the embedding time 

### ISSUE THREAD

**Issue by mikanbox** [langchain-ai/langchain-aws#188]
From the following code,  it seems that BedrockEmbedding is currently calling API sequentially based on the given list.

https://github.com/langchain-ai/langchain-aws/blob/4aebf1709d31220bcf248acdc9094f4547d6ef0c/libs/aws/langchain_aws/embeddings/bedrock.py#L164-L175

Typically, chunk sizes are not large, and the time required for embedding is also minimal, which results in a large number of API calls being executed sequentially. 
As a result, `the impact of network latency becomes significant and cannot be ignored`.

Additionally, due to differences in the availability of models by country, specifying a model that is only available in another country can further increase latency, causing the majority of the embedding time to be spent waiting for a response due to network latency.

The provided chunks are in a list, and by calling the embedding API in parallel, it should be possible to reduce the impact of latency.

**Comment by ravediamond** [comment]
Why not use aembed_documents ? It is an async call and will give far better performance.

**Comment by michaelnchin** [comment]
Revisiting this as there have been some recent developments lately.

For non-Cohere providers (Titan, Nova, etc.), `aembed_documents` is already optimized to execute `aembed_query` concurrently in its own thread for each text, which is fine since these providers don't differentiate between document and query type                                                  
                                                                                                                                                                                  
With the new fix in #881, `aembed_documents` is now improved for Cohere models: it now delegates to the sync `embed_documents` path, which takes advantage of Cohere's native support for submitting multiple texts per API request. Additionally, we use the correct `search_document` type over `search_query`, which Cohere does differentiate and optimize for.

TLDR; @ravediamond suggested, `aembed_documents` is the right approach here, as it provides concurrent execution for non-Cohere providers and batched requests for Cohere.
